homeassistant/components/recorder/models.py

```diff
@@ -0,0 +1,115 @@
+"""Models for SQLAlchemy."""
+
+import json
+from datetime import datetime
+
+from sqlalchemy import (Boolean, Column, DateTime, ForeignKey, Index, Integer,
+                        String, Text, distinct)
+from sqlalchemy.ext.declarative import declarative_base
+import homeassistant.util.dt as dt_util
+from homeassistant.remote import JSONEncoder
+
+# SQLAlchemy Schema
+# pylint: disable=invalid-name
+Base = declarative_base()
+
+
+class Events(Base):
+    # pylint: disable=too-few-public-methods
+    """Event history data."""
+
+    __tablename__ = 'events'
+    event_id = Column(Integer, primary_key=True)
+    event_type = Column(String(32), index=True)
+    event_data = Column(Text)
```

Comment:
I gave JsonEncodedDict a go, but it caused all sorts of strange issues regarding objects created in another thread, hanging tests, etc. Maybe file it as an improvement in Pivotal?